From 48713ba26af38a15a97fc7ebb0828cd287ef2447 Mon Sep 17 00:00:00 2001
From: Alexandre Rostovtsev <tetromino@gentoo.org>
Date: Fri, 20 Jul 2012 14:46:33 +0000
Subject: libtracker-fts: ICU cannot handle complex locale descriptions

ubrk_open expects the name of just a single locale (e.g. "en_US.UTF-8"),
not the full definition of your various locale variables and their
values as returned by glibc's setlocale(LC_ALL, NULL).

Instead, limit ourselves to LC_CTYPE, since after all, that's all we
need to determine word boundaries.

Fixes GB#675660.
---
diff --git a/src/libtracker-fts/tracker-parser-libicu.c b/src/libtracker-fts/tracker-parser-libicu.c
index 7388f69..69f75ed 100644
--- a/src/libtracker-fts/tracker-parser-libicu.c
+++ b/src/libtracker-fts/tracker-parser-libicu.c
@@ -676,7 +676,7 @@ tracker_parser_reset (TrackerParser *parser,
 
 		/* Open word-break iterator */
 		parser->bi = ubrk_open(UBRK_WORD,
-		                       setlocale (LC_ALL, NULL),
+		                       setlocale (LC_CTYPE, NULL),
 		                       parser->utxt,
 		                       parser->utxt_size,
 		                       &error);
--
cgit v0.9.0.2
